WordPress/gutenberg
# [RNMobile] Search Block: Move block toolbar options to settings panel

**[This PR](https://github.com/WordPress/gutenberg/pull/29500) must be merged before this PR can be taken out of draft.** 

**Gutenberg Mobile PR:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/3210

## Description

This PR moves the block toolbar options to the settings panel per the latest designs:

Main Settings | Button Options
-- | --
![Screen Shot 2021-03-05 at 8 02 26 PM](https://user-images.githubusercontent.com/5810477/110189680-db9aab80-7ded-11eb-86b2-05b35cc6e629.png)|![Screen Shot 2021-03-05 at 8 02 41 PM](https://user-images.githubusercontent.com/5810477/110189684-ddfd0580-7ded-11eb-9577-58825fafc9aa.png)

_Edit: Updated screenshots and code to match the latest i2 design_

**Design Note**:
- The styling has not been implemented in this PR. Styling will happen in a later PR after the designs have been finalized. This PR only moves the options to the settings panel.

Changes include:
- Move `Show search label`, `Button position`, and `Icon only button` options to the settings panel. 
- Remove old unused code. 

## How has this been tested?



Tested on both iOS simulator and Android emulator:

1. Add the search block to a post. Verify the block toolbar options for `hide search label`, `button position`, and `icon only button` no longer appear. 
2. Click the settings cog of the search block to open the settings panel. 
3. Verify the `Hide search heading` option is toggled off. Tap to toggle this option on and verify the search label disappears from the search block.
4.  Tap the `Button position` option and verify the following options appear: `Button inside`, `Button outside`, `No button`. Tap `Button inside`. Verify the label on the main settings panel shows `Inside` as the selected option. Also verify the UI updates appropriately. Verify the remaining two options.
5. Toggle the `Use button icon` option. Verify the Search block button changes to the icon button. 

## Screenshots 

https://user-images.githubusercontent.com/5810477/110189796-6aa7c380-7dee-11eb-937a-99d647406cf3.mov



## Types of changes




This PR is part of a new feature to port the Search Block to mobile and is one of many PRs that will make up the final feature. Currently this option is available in devMode only. 

## Checklist:
